advanced_samples/Bulk_Upload/DotNet/BulkUploader/Program.cs

```diff
@@ -0,0 +1,222 @@
+            }
+
+            if (ToThrow != null)
+                throw ToThrow;
+
+            Console.WriteLine("Success!!!");
+            return true;
+        }
+
+        private static void LogException(Exception ex)
+        {
+            Console.WriteLine(ex);
+            if (ToThrow == null)
+                ToThrow = ex;
+        }
+
+        private static void SendDataView()
+        {
+            string dataviewS = File.ReadAllText(DataviewPath);
+            List<DataView> dataviews = JsonConvert.DeserializeObject<List<DataView>>(dataviewS);
+            foreach (var dataview in dataviews)
+            {
+                DvService.CreateOrUpdateDataViewAsync(dataview).Wait();
+            }
+        }
+
+        private static void SendTypes()
+        {
+            Console.WriteLine($"Sending types from file: {SdsTypePath}");
+            string types = File.ReadAllText(SdsTypePath);
+            List<SdsType> typeList = JsonConvert.DeserializeObject<List<SdsType>>(types);
+            foreach (var type in typeList)
+            {
+                MetadataService.GetOrCreateTypeAsync(type).Wait();
+            }
+        }
+
+        private static void SendStreams()
+        {
+            Console.WriteLine($"Sending streams from file: {SdsStreamPath}");
+            string streams = File.ReadAllText(SdsStreamPath);
+            var streamsList = JsonConvert.DeserializeObject<List<SdsStream>>(streams); 
+            foreach (var stream in streamsList)
+            {
+                MetadataService.GetOrCreateStreamAsync(stream).Wait();
+
+                if (!string.IsNullOrEmpty(SdsStreamMetaPath))
+                {
+                    try
+                    {
+                        string path = SdsStreamMetaPath + stream.Id + ".json";
+                        Console.WriteLine($"Sending stream metadata from file: {path}");
+                        string meta = File.ReadAllText(path);
+                        MetadataService.UpdateStreamMetadataAsync(stream.Id, JsonConvert.DeserializeObject<IDictionary<string, string>>(meta));
+                    }
+#pragma warning disable CA1031 // Do not catch general exception types
+                    catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+                    {
+                        LogException(ex);
+                    }
+                }
+
+                if (!string.IsNullOrEmpty(SdsStreamTagPath))
+                {
+                    try
+                    {
+                        string path = SdsStreamTagPath + stream.Id + ".json";
+                        Console.WriteLine($"Sending stream tag from file: {path}");
+                        string meta = File.ReadAllText(path);
+                        MetadataService.UpdateStreamTagsAsync(stream.Id, JsonConvert.DeserializeObject<IList<string>>(meta));
+                    }
+#pragma warning disable CA1031 // Do not catch general exception types
```

Comment:
I don't have a problem with this but just FYI, the way I've done this for other samples is to use an attribute on the function, like here: https://github.com/osisoft/OSI-Samples-OCS/blob/Bulk-Upload/basic_samples/SDS/DotNet/SdsClientLibraries/SdsClientLibraries/Program.cs#L22

Reply:
i actually got rid of these, most of the try catches are weird and if it breaks it should probably be dead immediately, but i updated the ones that make sense to me to this